keras_cv/models/segmentation/fcn.py

```diff
@@ -0,0 +1,646 @@
+                    )
+                backbone_name = backbone.name.lower()
+                output_tensor = vgg_architecture_builder(
+                    classes=classes,
+                    backbone=backbone,
+                    model_architecture=model_architecture,
+                    input_tensor=input_tensor,
+                    input_shape=input_shape,
+                )
+
+                del backbone
+
+                data_format = tf.keras.backend.image_data_format()
+                if return_mask:
+                    if data_format == "channels_last":
+                        output_tensor = tf.math.argmax(output_tensor, axis=3)
+                        output_tensor = tf.expand_dims(output_tensor, axis=3)
+                    else:
+                        output_tensor = tf.math.argmax(output_tensor, axis=1)
+                        output_tensor = tf.expand_dims(output_tensor, axis=1)
+
+                output_tensor = tf.cast(output_tensor, tf.dtypes.float32)
+
+                super().__init__(
+                    inputs={"input_tensor": input_tensor},
+                    outputs={"output_tensor": output_tensor},
+                )
+
+                self.classes = classes
+                self.backbone = backbone_name
+                self.model_architecture = model_architecture
+                self.return_mask = return_mask
+
+        elif isinstance(backbone, tf.keras.models.Model) or isinstance(
+            backbone, tf.keras.Functional
+        ):
+            if model_architecture is not None:
+                raise ValueError(
+                    """`model_architecture` cannot be set if `backbone` is not a `keras_cv.models.VGG16` or
+                    `keras_cv.models.VGG19`. Either set `backbone` to one of the accepted values or remove
+                    the `model_architecture` argument."""
+                )
+            else:
+                if len(input_shape) != 4:
+                    input_tensor = utils.parse_model_inputs(input_shape, input_tensor)
+                else:
+                    input_tensor = tf.keras.layers.Input(
+                        shape=(input_shape[1], input_shape[2], input_shape[3])
+                    )
+                backbone = custom_architecture_builder(backbone)
+                classes_conv = tf.keras.layers.Conv2D(
+                    filters=classes,
+                    kernel_size=(1, 1),
+                    strides=(1, 1),
+                    padding="same",
+                    activation="softmax",
+                )
+
+                input_shape = input_tensor.shape
+
+                x = backbone(input_tensor)
+
+                output_shape = x.shape
+
+                if input_shape[0] is None and input_shape[1] is None:
+                    if None in backbone.input.shape:
+                        raise ValueError(
+                            f"""No `input_shape` argument available to model. `input_shape`
```

Comment:
Docstring -> string

Reply:
Okay, will do. I mainly converted them for linting purposes, though

Reply:
Multi-line string shouldn't flare up the linter, afaik